segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ReadOnlySegmentContainer.java

```diff
@@ -197,6 +198,11 @@ public boolean isOffline() {
         return unsupported("truncateStreamSegment");
     }
 
+    @Override
+    public CompletableFuture<DirectSegmentAccess> forSegment(String streamSegmentName, Duration timeout) {
```

Comment:
1. Since the ROSS has been around for more than 1 year, I haven't seen a single user trying to access it or request improvements. Until that happens, I will keep it in Beta and will only support existing features on it, and not spend time to add new ones.
2. Direct Segment Access is for internal components only, not for external access. ROSS is for external access only and adding Direct Segment Access will never benefit it. Hence this is unlikely to ever be implemented in this one.

Reply:
> Direct Segment Access is for internal components only, not for external access. ROSS is for external access only and adding Direct Segment Access will never benefit it. Hence this is unlikely to ever be implemented in this one.

Direct access enables multiple sequential reads over the same segment. Can't a client of the ROSS benefit from it? I think that if there is at least a chance that we implement such a functionality for the ROSS, then we need to create an issue so that we track any potential future work.